### PR TITLE
fix mail sync and purchase invoice upload

### PR DESCRIPTION
**Pull request by patrickweh**
fix invalid media attachments in CreateMailMessage.php

add possibility to pass existing media id to CreatePurchaseInvoice.php

add possibility to add mime type on media upload

### CODE REVIEW

File: src/Jobs/SyncMailAccountJob.php [hunk 1]
```diff
@@ -60,15 +62,13 @@ public function handle(): void
...
-                fn (MailFolder $folder) => DeleteMailFolder::make(['id' => $folder->id])->validate()->execute()
-            );
+            ->each(fn (MailFolder $folder) => DeleteMailFolder::make(['id' => $folder->id])->validate()->execute());
 
         if ($this->onlyFolders) {
             return;
         }
 
-        foreach ($folders as $folder) {
+        foreach ($folders->reverse() as $folder) {
```
Comment (SirSplasch) [inline src/Jobs/SyncMailAccountJob.php:71]: why reverse?

File: src/Jobs/SyncMailAccountJob.php
Comment (SirSplasch) [inline src/Jobs/SyncMailAccountJob.php:1]: Resolve all Model (queries) via app container app(Model::class)->query()

File: src/Jobs/SyncMailAccountJob.php [hunk 2]
```diff
@@ -126,8 +126,8 @@ private function getNewMessages(Folder $folder): void
                 ->get()
                 ->first()
                 ?->getUid() - 1
```
Comment (SirSplasch) [inline src/Jobs/SyncMailAccountJob.php:128]: null - 1 == true | this leads to unexpected behaviour

File: src/Actions/PurchaseInvoice/CreatePurchaseInvoice.php [hunk 3]
```diff
@@ -64,8 +73,26 @@ protected function validateData(): void
         parent::validateData();
 
         $errors = [];
+        $media = data_get($this->data, 'media');
+        $filePath = (
```
Comment (SirSplasch) [inline src/Actions/PurchaseInvoice/CreatePurchaseInvoice.php:77]: why not use match (true) {}?

**Review by SirSplasch — CHANGES_REQUESTED** [review]

File: src/Jobs/SyncMailAccountJob.php (hunk 1 above)
Comment (patrickweh) [reply src/Jobs/SyncMailAccountJob.php:71]: the imap package fetches in reversed order compared to any regular client, so the inbox is at bottom not at top

**Review by SirSplasch — APPROVED** [review]

**Pull request merged**